src/reporter.ts

```diff
@@ -9,10 +9,14 @@ import {
 import {Octokit} from '@octokit/rest'
```

Comment:
I'm not sure if it's worth to check to change to [`@octokit/action`](https://github.com/octokit/action.js) here rather than implementing it here again since it already implements proxy support: https://github.com/octokit/action.js/pull/329

Reply:
As is the case with a lot of proxy implementations it appears to be a bit "basic".   Will raise an issue there with the details and see what they say as well.

Reply:
@stoe just for my understanding, is the proxy handling something that is always needed for Actions in GHES instances? Are the environment variables always the same or can they differ/be set by the customer? I'm happy to use the superior `proxy-agent` dependency in `@octokit/action`, I just want to fully understand the context before making the change.

Reply:
@gr2m, they won't be always needed as this depends on the implementation, but when used they should always follow our documented pattern under https://docs.github.com/en/actions/hosting-your-own-runners/using-a-proxy-server-with-self-hosted-runners

Reply:
Also, this is independent of GHES and would also apply for self-hosted runners on GHEC (github.com).

Reply:
Got it, thank you for clarifying. As it's documented in the GitHub docs I consider it best practice, and will implement it for the Action SDK (`@octokit/action`). I think you should be able to safely use it instead of `@octokit/rest`. I'm happy to help with the transition.

I do plan a bigger overhaul of `@octokit/action` (to make it behave more like `@octokit/app`) but that won't happen any time soon and when we get there, I'll help you with the upgrade as well.